server/bleep/src/indexes/file.rs

```diff
@@ -791,29 +899,34 @@ impl RepoFile {
                         lang_str,
                         &self.branches,
                     )
-                    .await;
+                    .await
             })
         });
 
-        Some(doc!(
-            schema.raw_content => buffer.as_bytes(),
-            schema.raw_repo_name => repo_name.as_bytes(),
-            schema.raw_relative_path => relative_path_str.as_bytes(),
-            schema.unique_hash => cache_keys.tantivy(),
-            schema.repo_disk_path => repo_disk_path.to_string_lossy().as_ref(),
-            schema.relative_path => relative_path_str,
-            schema.repo_ref => repo_ref.to_string(),
-            schema.repo_name => *repo_name,
-            schema.content => buffer,
-            schema.line_end_indices => line_end_indices,
-            schema.lang => lang_str.to_ascii_lowercase().as_bytes(),
-            schema.avg_line_length => lines_avg,
-            schema.last_commit_unix_seconds => last_commit,
-            schema.symbol_locations => bincode::serialize(&symbol_locations).unwrap(),
-            schema.symbols => symbols,
-            schema.branches => branches,
-            schema.is_directory => false,
-            schema.indexed => true,
+        stats.chunks += insert_stats.new;
+
+        Some((
+            stats,
```

Comment:
I think this may be cleaner if the workload had a `tx` reference, and instead of bubbling these back up, the event was generated and sent off at this point.

Reply:
agreed. have added a `tx` to each workload that `transmit_stats` when available.